### ISSUE TITLE
About dataset

### ISSUE THREAD

**Issue by seunghee-han** [bcmi/TopNet-Object-Placement#1]
Hello, have read your paper, and I think your paper is very good.
It seems that "data.rar" is missing a few files.
Can i get "train(test)_pair_new.json" and "train(test)_pair_new.csv" file?

**Comment by WhynotHAHA** [comment]
We have submitted the required "train(test)_pair_new.json" and "train(test)_pair_new.csv" files.

**Comment by seunghee-han** [comment]
Thank You.
I want to train Topnet.
So Can i get "network" folder?

**Comment by WhynotHAHA** [comment]
OK, I have submitted "network" folder.

**Comment by seunghee-han** [comment]
I am training a Topnet model.
However, I cannot find the code for heatmap. 
And fg data and bg data are input into the transformer using the same vit encoder in "network_vit.py". So how do the global and local features come out? 
please tell me. Thank You

**Comment by WhynotHAHA** [comment]
You can refer to the function "fopa_heat_map" of Libcom (https://github.com/bcmi/libcom/tree/main/libcom) for generating  heatmap. It is worth noting that you need to replace the network in Libcom from FOPA to TopNet. And I recommend you to use ResNet encoder in "network.py" for better performance.

**Comment by seunghee-han** [comment]
Can I get the 'Sparse Contrastive Loss' code written in the paper? 
Your code uses CrossEntropyLoss.
Thank you.

**Comment by seunghee-han** [comment]
I am very interested in the TopNet paper.
So I want to take a closer look at the code. 
Can I get the code with sparse contrastive loss?
I wish the heatmap code was included as well. 
I would really like it if the complete train.py and test.py file were uploaded.
Please tell me. Thank you

**Comment by ustcnewly** [comment]
Hi, seunghee-han, it is worth noting that TopNet is not our paper and this repo is just an unofficial implementation.  Actually, the code is adapted from the implementation of our FOPA. It uses both positive and negative labels in OPA dataset, which is different from the losses used in TopNet paper.

**Comment by WhynotHAHA** [comment]
1. After you run the train.py file to get the trained model, you can run the test.py file to get the F1 score and balanced accuracy of the model on the OPA test set.
2. Refer to https://github.com/bcmi/libcom/tree/main to complete the installation of the corresponding environment. Place the trained model in the /libcom/libcom/fopa_heat_map/pretrained_models folder. Replace the /libcom/libcom/fopa_heat_map/fopa_heat_map.py file with network.py in TopNet. Change /libcom/libcom/fopa_heat_map/__init__.py.
3. Python /libcom/tests/test_fopa_heat_map.py，then you can get selected object placements, heatmaps, different scaled foregrounds.
4. You can randomly sample one from 50 generated composite images per background-foreground pair for Acc and FID evaluation, using the test scripts provided by [GracoNet](https://github.com/bcmi/GracoNet-Object-Placement).

I recommend you use FOPA as an example to familiarize yourself with the heatmap generation process in Libcom, and then just replace network.py and pretrained model to complete TopNet's heatmap generation. 
Because we used OPA's positive labels and negative labels in our reproduction process to facilitate comparison with other models, we did not follow the sparse contrastive loss in the original paper but used cross entropy.
